File: wms-fe/src/services/warehouseService.js
import axiosClient from '../api/axiosClient';

const warehouseService = {
  getAll: (params) => {
    return axiosClient.get('/warehouses', { params });
  },
  
  getById: (id) => {
    return axiosClient.get(`/warehouses/${id}`);
  },
  
  create: (data) => {
    return axiosClient.post('/warehouses', data);
  },
  
  update: (id, data) => {
    return axiosClient.put(`/warehouses/${id}`, data);
  },
  
  delete: (id) => {
    return axiosClient.delete(`/warehouses/${id}`);
  }
};

export default warehouseService;

File: wms-be/src/main/java/com/restapi/wmsservice/service/WarehouseService.java
package com.restapi.wmsservice.service;

import com.restapi.wmsservice.dto.request.WarehouseRequest;
import com.restapi.wmsservice.dto.response.WarehouseResponse;

import java.util.List;

public interface WarehouseService {
    WarehouseResponse create(WarehouseRequest request);
    List<WarehouseResponse> getAll();
    WarehouseResponse getById(Long id);
    WarehouseResponse update(Long id, WarehouseRequest request);
    void delete(Long id);
}
